src/roles/netdata/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+
+## Add test to see if it's already installed and skip if it is
+- name: Check if Netdata is present
+  stat:
+    path: /opt/netdata
+  register: netdata_present
+
+- debug: msg="netdata not installed"
+  when: netdata_present.stat.exists == False
+
+- debug: msg="netdata installed"
+  when: netdata_present.stat.exists == True
+
+## It will be running once installed
+- name: Install Intel/AMD 64bit static build of Netdata
+  shell: bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh) --dont-wait
+  args:
+    executable: /bin/bash
+  become: no
+  when: netdata_present.stat.exists == False
+
+- name: Ensure port 20000 allowed for list of servers
+  include_role:
+    name: firewall_port
+  vars:
+    firewall_action: open
+    firewall_port: 20000
+    firewall_protocol: tcp
+# Don't define this, or the rule won't get added to localhost
+#    firewall_servers: "{{ groups['app-servers'] }}"
+    firewall_zone: "{{m_private_networking_zone|default('public')}}"
+
+- name: Configure HAProxy to allow Netdata
+  blockinfile:
```

Comment:
Don't don `blockinfile` for `haproxy.cfg`. Just edit `src/roles/haproxy/templates/haproxy.cfg.j2` and add the following block inside `if load_balancer_handle_external`:

```
{% if m_install_netdata %}
frontend netdata
	bind *:20000 ssl crt {{ ssl_certificate_file }}
	mode http
	default_backend netdata-back
backend netdata-back
	server nd1 127.0.0.1:19999
{% endif %}
```